File: swiftPrep/Sources/Sorting/SelectionSort.swift
func selectionSort(sequence: inout [Int]) {
	for outerIndex in 0..<sequence.count {
		var minIndex = outerIndex

		for innerIndex in outerIndex+1..<sequence.count {
			if sequence[innerIndex] < sequence[minIndex] {
				minIndex = innerIndex
			}
		}

		(sequence[outerIndex], sequence[minIndex]) = (sequence[minIndex], sequence[outerIndex])
	}
}

File: javascriptPrep/lib/sorting/selectionSort.js
#!/usr/bin/env node

"use strict";

function selectionSort(seq) {
	for (let outerIndex = 0; outerIndex < seq.length; outerIndex++) {
		let minIndex = outerIndex;

		for (let innerIndex = outerIndex+1; innerIndex < seq.length; innerIndex++) {
			if (seq[innerIndex] < seq[minIndex]) {
				minIndex = innerIndex;
			}
		}

		[seq[outerIndex], seq[minIndex]] = [seq[minIndex], seq[outerIndex]];
	}
}

module.exports = selectionSort;